### ISSUE TITLE
noscript contents are ending up in the output

### ISSUE THREAD

**Issue by jribbens** [xi/aria-api#7]
aria-api should be ignoring the contents of `<noscript>` elements. These elements turn their contents into text nodes, even though their contents is in fact HTML. So the accessibility name/description end up being a load of gibberish HTML for any element that contains a noscript element.

**Comment by xi** [comment]
Can you provide a reproducible example? I tried the following, both of which had the correct result:

- [`<noscript>test</noscript>`](https://xi.github.io/babelacc/?input=%3Cnoscript%3Etest%3C%2Fnoscript%3E) -> ""
- [`<span aria-labelledby="foo"></span> <noscript id="foo">test</noscript>`](https://xi.github.io/babelacc/?input=%3Cspan+aria-labelledby%3D%22foo%22%3E%3C%2Fspan%3E%0D%0A%3Cnoscript+id%3D%22foo%22%3Etest%3C%2Fnoscript%3E) -> "test"

**Comment by jribbens** [comment]
Calling `getName()` on `#x` below should return `test` but returns literally `test <span>moo</span>`.
```html
<a id="x" href="#">test <noscript><span>moo</span></noscript></a>
```
(Chrome and Firefox get this one right. Example test page at https://unequivocal.eu/aria.html)

**Comment by jribbens** [comment]
I just realised your links above are to some sort of comparison page for different implementations, and they *all* get this wrong except `axe`. I guess this must be due to a faulty specification. But there is no possible way of looking at this whereby putting raw HTML in the 'accessible name' makes any sense ;-)

**Comment by xi** [comment]
From what I understand, there are two angles to this.

From a theoretical perspective, `<noscript>` works [very differently depening on whether scripting is enabled ir not](https://html.spec.whatwg.org/multipage/scripting.html#the-noscript-element): If scripting is enabled, the content of a `<noscript>` element are interpreted as a single text node. If scripting is disabled, the content is interpreted as HTML. As this is a javascript library it only makes sense to consider the case that scripting is enabled. In that case the behavior is correct: The angle brackets are just part of the content and should not be interpreted as HTML.

 From a practical perspective I completely agree with you: including HTML in an accessible name is pointless. However, also from a practical perspective, I wonder whether this is really a relevant case, considering that the algorithm is already very complicated.

I think it would be a good idea if you file an issue with the spec and also include a real-world use case where this is relevant.

I will also try to find a short term solution for this library. I guess the simplest and best solution would be to ignore `<noscript>` elements completely, right?

**Comment by jribbens** [comment]
OK I have created an issue with the spec: w3c/accname#134

This absolutely is relevant to the real world - if you don't fix it I will have to fork your project and fix it myself for my usage! The first real-world page I encountered this problem on is here: https://www.omag.org/ (multiple occurrences on this page, for example the link that contains the "police cruiser" image above the text "General liability, auto & equipment").

**Comment by xi** [comment]
I just realized why I was not able to reproduce this: This is an issue with chromium, but not with firefox. In firefox `<noscript>` defaults to `display: none` while in chromium it defaults to `display: inline`.

According to step (2) in the algorithm, hidden elements must not be included in the accessible name. I guess `<noscript>` should always be interpreted as hidden (as long as script is enabled). So far the code only checks for `display` and `visibility`, so in chromium `<noscript>` was reported as not-hidden.

It may be worth the time to file an issue with chromium. I am not sure.

A fix should be simple, I just have to finish up some tests.
